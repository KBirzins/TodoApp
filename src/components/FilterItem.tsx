import React from 'react';
import {TouchableWithoutFeedback} from 'react-native';
import styled from 'styled-components/native';
import Checkbox from './common/CheckBox';

const ContainerView = styled.View`
  flex-direction: row;
  align-items: center;
  padding-vertical: 5px;
`;

const StyledText = styled.Text`
  margin-left: 5px;
`;

interface IFilterItem {
  isFilterActive: boolean;
  setFilterActive(arg0: boolean): any;
  filterType: boolean;
  setFilterType(arg0: boolean): any;
  primaryText: string;
  secondaryText: string;
}

const FilterItem = ({
  isFilterActive,
  setFilterActive,
  filterType,
  setFilterType,
  primaryText,
  secondaryText,
}: IFilterItem) => {
  return (
    <ContainerView>
      <Checkbox
        selected={isFilterActive}
        onPress={() => setFilterActive(!isFilterActive)}
      />
      <TouchableWithoutFeedback
        onPress={() => {
          if (filterType) {
            setFilterType(false);
          } else {
            setFilterType(true);
          }
        }}>
        {filterType ? (
          <StyledText>{primaryText}</StyledText>
        ) : (
          <StyledText>{secondaryText}</StyledText>
        )}
      </TouchableWithoutFeedback>
    </ContainerView>
  );
};

export default FilterItem;
