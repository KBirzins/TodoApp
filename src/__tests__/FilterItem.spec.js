import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import FilterItem from '../components/FilterItem';

describe('FilterItem', () => {
  it('renders correctly with correct properties', () => {
    const component = renderer.create(<FilterItem />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
