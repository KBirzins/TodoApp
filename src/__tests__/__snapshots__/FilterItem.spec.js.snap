// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem renders correctly with correct properties 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingVertical": 5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "color": "#211f30",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î µ
  </Text>
  <Text
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "marginLeft": 5,
        },
      ]
    }
  />
</View>
`;
